src/swsssdk/interface.py

```diff
@@ -359,6 +359,6 @@ def _connection_error_handler(self, db_id):
         In the event Redis is unavailable, close existing connections, and try again.
         """
         logger.warning('Could not connect to Redis--waiting before trying again.')
-        self.close(db_id)
+        self._close(db_id)
         time.sleep(self.CONNECT_RETRY_WAIT_TIME)
-        self.connect(db_id, True)
+        self._connect(db_id, True)
```

Comment:
>_connect [](start = 13, length = 8)

We need to fix SonicV2Connector. There is nothing wrong in DBInterface

Reply:
use composition instead and only change SonicV2Connector wrapper.